examples/random.h

```diff
@@ -0,0 +1,31 @@
+#if defined(_WIN32)
+#include <bcrypt.h>
+#elif defined(__linux__)
+#include <fcntl.h>
+#include <unistd.h>
+#else
+#error "Couldn't identify the OS"
+#endif
+
+
+/* Returns 1 on sucess, and 0 on failure. */
+int fill_random(unsigned char* data, unsigned long size) {
+#if defined(_WIN32)
+    NTSTATUS res = BCryptGenRandom(NULL, data, size, BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+    if (res != STATUS_SUCCESS) {
+        return 0;
+    }
+#elif defined(__linux__)
+    ssize_t res;
+    int fd = open("/dev/urandom", O_RDONLY);
```

Comment:
Would be better if we'd use getrandom if available.

Reply:
I really wanted to use `getrandom` but it's not available in the libc in our travis.
We could add `dist: bionic` and that will probably solve this, but my thought was 
"if even our travis doesn't support it then we should probably not use this in the examples"

See: https://travis-ci.org/github/elichai/secp256k1/jobs/681424984#L403

Reply:
Good point. We can update this later.

Reply:
Oh weird, it's really not in Ubuntu LTS 16.04. You need glibc 2.25 at least. -.- I wanted to point out `getrandom` too...

I think we should at least add an explanatory comment or a link to an explanation. And to be honest, I'd prefer to use `getrandom` here and have an example that breaks compilation if it's not available and refer people to `/dev/urandom` then.

Reply:
I thought about that, but how do you propose to do this?
like should I involve autotools in this now?(to detect getrandom) I really preferred to keep the examples minimal and self standing.

another advantage of `/dev/urandom` is that I can just stick MacOS and BSD variants on it and it should work

Reply:
> like should I involve autotools in this now?(to detect getrandom) I really preferred to keep the examples minimal and self standing.

Hm ok, what I had in mind was rather not compiling the examples at all in our build system. 
But I see that this is not great either, too.